R/tfb-wavelet-utils.R

```diff
@@ -29,28 +29,52 @@ interpolate_arg <- function(arg_list) {
 
 fit_wavelet <- function(data, Z, penalized, glmnet_args) {
   eval_list <- split(data$value, data$id)
-  arg_list <- split(data$arg, data$id)
+  arg_list <- unique(data$arg)
+  
+  Z <- cbind(Z, arg_list)
+  Z <- scale(Z, center = FALSE)
   
-  correction <- map2(arg_list, eval_list, function(x, y) remove_slope(x,y))
+  if (!penalized) {
+    coefs <- map(eval_list, 
+                  function(x) {
+                    # Least squares fit, directly computed
+                    unname(lsfit(Z, x, intercept = TRUE)$coef)})
```

Comment:
I use lsfit for the least squares fit now. Also Z is now scaled after everything is included and the grid is also included in the fit as opposed to before where it was added afterwards.

Reply:
note to self: change to qr.coef eventually. do single matrix solve for regular data.